site/docs/hosting/zeabur-nodejs.md

```diff
@@ -0,0 +1,132 @@
+---
+prev: false
+next: false
+---
+
+# Hosting: Zeabur (Node.js)
+
+[Zeabur](https://zeabur.com) is a platform that allows you to deploy your full-stack applications with ease. It supports various programming languages and frameworks, including Node.js and grammY.
+
+This tutorial will guide you how to deploy your grammY bots with Node.js to [Zeabur](https://zeabur.com/).
+
+::: tip Looking for the Deno Version?
+This tutorial explains how to deploy a Telegram bot to Zeabur using Node.js.
+If you're looking for the Deno version, please check out [this tutorial](./zeabur-deno) instead.
+:::
+
+## Prerequisites
+
+To follow along, you need to have [Github](https://github.com) and [Zeabur](https://zeabur.com/) accounts.
+
+### Method 1: Create a New Project From Scratch
+
+Initialize your project and install some necessary dependencies:
+
+```sh
+# Initialize the project.
+mkdir grammy-bot
+cd grammy-bot
+pnpm init -y
+
+# Install main dependencies.
+pnpm install grammy
+
+# Install development dependencies.
+pnpm install -D typescript
+
+# Initialize TypeScript.
+npx tsc --init
+```
+
+Then, cd into `src/`, and create a file named `bot.ts`.
+It is where you will write your bot's code.
+
+Now, you can start writing your bot's code in `src/bot.ts`.
+
+```js
+import { Bot } from "grammy";
+
+const bot = new Bot(
+  process.env.TELEGRAM_BOT_TOKEN || "your_telegram_bot_token",
+);
+
+bot.on("message:text", async (ctx) => {
+  console.log("Message: ", ctx.message.text);
+
+  const response = "Hello, I'm a bot!";
+
+  ctx.reply(response);
+});
+
+bot.start();
+```
+
+> Note: Get your bot token with [@BotFather](https://t.me/BotFather) on Telegram, and set is as an environment variable `TELEGRAM_BOT_TOKEN` in Zeabur.
+>
+> You can check out [this tutorial](https://zeabur.com/docs/deploy/variables) for setting environment variables in Zeabur.
+
+Now your project's root directory should now look like this:
+
+```asciiart:no-line-numbers
+.
+├── node_modules/
+├── src/
+│   └── bot.ts
+├── package.json
+├── pnpm-lock.yaml
+```
+
+And then we have to add `start` scripts to our `package.json`.
+Our `package.json` should now be similar to this:
+
+```json
+{
+  "name": "telegram-bot-starter",
+  "version": "1.0.0",
+  "description": "Telegram Bot Starter with JavaScript and Grammy",
+  "scripts": {
+    "start": "ts-node src/bot.ts"
```

Comment:
This cannot work if `ts-node` is not installed?

Reply:
oh my bad for not considering this situation, gonna fix it soon

Reply:
Just updated the doc, can u help check again? :D Thanks for the patient review!